### ISSUE TITLE
🐛 BUG: Relative image URL in Markdown file breaks the build

### ISSUE THREAD

**Issue by WaltzingPenguin** [snowpackjs/astro#1188]
### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

I have a route for displaying each page in a blog:

```
---
export async function getStaticPaths () {
  const data = Astro.fetchContent('../../posts/**/*.md')
  return data.map(post => ({
    params: {slug: post.slug},
    props: { post }
  }))
}

const { post } = Astro.props
---

{ post.astro.html }
```

If any loaded markdown file includes a relative image link, e.g. `![](./logo.svg)`, the build fails (no files are written) with the error:
```
[build] ! building pages...
[relative-link] This value will be resolved relative to the page: <img src="./logo.svg">
[build] ✔ pages built.
[build] file:///C:/www-blog/src/pages/post/[slug].astro: could not find "logo.svg"
```

### Steps to Reproduce

1. Checkout https://github.com/WaltzingPenguin/astro-bug-relative-image-url
2. `npm install`
3. `npm run build`
4. Observe error

### Link to Minimal Reproducible Example (Optional)

https://github.com/WaltzingPenguin/astro-bug-relative-image-url

**Comment by tony-sull** [comment]
Thanks for filing a repo with a clean reproduction!

This actually looks like Astro is worked as expected here.  `logo.svg` is saved at `/public/post/first` but with the relative import Astro is actually expecting to find it at `src/posts`

If you want to keep the SVG in the `public` directory you just need to reference as `![](/post/first/logo.svg)`

**Comment by WaltzingPenguin** [comment]
> This actually looks like Astro is worked as expected here. `logo.svg` is saved at `/public/post/first` but with the relative import Astro is actually expecting to find it at `src/posts`

I would love for that to be the case.  It is not, however.  The build fails with the same error.

**Comment by snelgrove** [comment]
I've just updated a project to the latest release and found a similar problem. I cloned the repo in the first post and updated it to `astro@0.20.9` and it looks like we have the same issue, so I thought I'd add some detail. 

Upon updating, here are the differences I observed between these astro releases:

- In `astro@0.18`, images could be included in `/src/posts/` directory, and referenced in markdown using a relative path
- In `astro@0.20.9`
  - Relative paths in markdown files now reference the `/public` directory
  - We can no longer include images under `/src/posts/`, as astro will attempt to find and build `image.png.js`, causing a build error

So, I moved my images to `/public`, which led me to this issue. To get the example repo working in both `dev` and `build`, I found that I have to use absolute paths to images, which I don't think is the intended behaviour:

- Absolute paths can be used, and they reference files in `/public`. For example `[](/post/first/logo.svg)`✅
- In `astro dev`, the relative path `[](first/logo.svg)`, in `/src/posts/first.md` maps to `/public/post/first/logo.svg` ✅
- In `astro build`, the relative path `[](first/logo.svg)` causes a build error. The  compiler resolves the relative path to `/_astro/src/pages/post/first/logo.svg` rather than the `/public` folder 🚫

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
